src/main/core/work/task.rs

```diff
@@ -0,0 +1,140 @@
+use std::sync::Arc;
+
+use atomic_refcell::AtomicRefCell;
+
+use crate::{core::worker::Worker, host::host::Host};
+
+/// Mostly for interoperability with C APIs.
+/// In Rust code that doesn't need to interact with C, it may make more sense
+/// to directly use a `FnMut(&mut Host)` trait object.
+pub struct Task {
+    inner: Arc<AtomicRefCell<dyn FnMut(&mut Host)>>,
```

Comment:
Yeah, I think normally in Rust code it'd be better for the `Arc` and probably the `AtomicRefCell` to be explicit. For interoperation with C though we need to wrap it all up.

One possibility I considered is to have a `Task` that basically just stores the function (and maybe the `magic`), and call the thing we expose to C something like `TaskRef`. Also I think in that case we might be able to just have something like `type TaskRef = Arc<AtomicRefCell<Task>>`. Wdyt?

> But in https://github.com/shadow/shadow/pull/2143 you removed the Arc anyways, so this is a non-issue, but still wanted to mention it here anyways.

True, but I'm not 100% sure that's going to work out yet; there was a leak I wasn't able to track down last night, but maybe I'll have better luck today with fresh eyes :). Still, I'd rather make sure we're fairly happy with merging this as-is whether or not the follow-up ends up happening.

Reply:
> Also I think in that case we might be able to just have something like `type TaskRef = Arc<AtomicRefCell<Task>>`. Wdyt?

I think this is the best solution. I've done it this way before (I can't remember where though) and I think it worked out well. I've also [cheated](https://github.com/shadow/shadow/blob/f793e086b6b8170dc284c579ba4394b6b970a6e4/src/main/host/host.c#L78) and used the full name `Arc_AtomicRefCell_AbstractUnixNamespace`, but only because it was used in only 2 or 3 places in the C code.